### ISSUE TITLE
sql: migration run but not recorded

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#20928]
multiple nodes on `cyan` crashed with `failed to make root a member of the admin role: duplicate key value (role,member)=('admin','root') violates unique constraint "primary"`.

This is the usual rolling restart performed by continuous deployment.
Deployment timeline on node 1 (the first one to be restarted):
I171219 23:01:41:  62a358c last sha without the migration
I171220 00:01:44: e9f1851 (first sha with the migration)
I171220 05:03:09: 7957839 (some other sha)
I171220 13:02:39: faf9cbd (crashing sha)

All nodes are full of logs looking more or less like:
```
I171220 05:03:26.467625 38 server/server.go:1226  [n1] done ensuring all necessary migrations have run
W171220 13:02:54.926201 16 sqlmigrations/migrations.go:829  [n1] failed to make root a member of the admin role: duplicate key value (role,member)=('admin','root') violates unique constraint "primary"
W171220 13:02:55.082658 16 sqlmigrations/migrations.go:829  [n1] failed to make root a member of the admin role: duplicate key value (role,member)=('admin','root') violates unique constraint "primary"
W171220 13:02:55.284888 16 sqlmigrations/migrations.go:829  [n1] failed to make root a member of the admin role: duplicate key value (role,member)=('admin','root') violates unique constraint "primary"
```

**Comment by a-robinson** [comment]
All it would take is one node (or user) having successfully run the migration once without persisting the fact that it had done so to get all nodes into this state permanently. So #20926 is definitely needed regardless of the root cause of how the cluster got here -- all migrations need to be idempotent.

**Comment by a-robinson** [comment]
So while I'm still curious how the cluster got into this state, since it isn't a common or likely occurrence, but it is a perfectly valid state for the cluster to be in.

**Comment by mberhault** [comment]
Ok, clearly my understanding of migrations was a little short.
I'll keep open until it recovers properly once #20926 is deployed.

**Comment by mberhault** [comment]
Cyan picked up #20926 and went back to normal. Other potential migration replay problem (not seen on cyan) fixed in #20966.
